File: check_db.py
import asyncio
import asyncpg
import os
from dotenv import load_dotenv

async def check_db():
    load_dotenv()
    dsn = os.getenv("DATABASE_URL")
    print(f"Connecting to: {dsn}")
    
    try:
        conn = await asyncpg.connect(dsn)
        rows = await conn.fetch("SELECT guild_id FROM guild_settings")
        print(f"\n--- DATABASE DIAGNOSTIC ---")
        print(f"Count of guilds in DB: {len(rows)}")
        print("Guild IDs found:")
        for row in rows:
            print(f" - {row['guild_id']}")
        print(f"---------------------------\n")
        await conn.close()
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    asyncio.run(check_db())

File: web/check-db.js
const { Pool } = require('pg');
require('dotenv').config({ path: '.env.local' });

async function checkDb() {
  console.log("Connecting to:", process.env.DATABASE_URL);
  const pool = new Pool({ connectionString: process.env.DATABASE_URL });
  
  try {
    const res = await pool.query('SELECT guild_id FROM guild_settings');
    console.log("Successfully retrieved data!");
    console.log("Count of guilds in DB:", res.rows.length);
    console.log("Guild IDs in DB:", res.rows.map(row => row.guild_id));
  } catch (err) {
    console.error("Database connection error:", err.message);
  } finally {
    await pool.end();
  }
}

checkDb();
